Cart Rules + Catalog Price Rules (#246)
PrestaShop/prestashop-specs: content/1.7/back-office/catalog/discount/add-edit-cart-price-rule.md

---
title: Adding and Editing cart rules
weight: 1
---

# Cart Rule > Edit
When you enter the Cart rule editing UI, there are 3 separate tabs - Information, Conditions and Actions. Each of them are combined together with the general cart rules functionality.

The explanations of the input fields, dropdowns, switch toggle buttons and text fields are stated below:<br>

## Information tab

![Cart Rules: Information tab](/static/img/cart-rules-information-tab.png)

> - **Name** - it is a mandatory field, it can be set as unique rule name. There is a widget at the right of this field with multiple languages. The different languages can be set to the rule names for this certain Information tab form.

Reviewer: ```suggestion
- **Name** - text field - it is a mandatory field with a language selector
```

> - **Description** - it can be described where is the cart rule used for in more details.

Reviewer: ```suggestion
- **Description** - text field - to describe the cart rule in more details.
```

> - **Code** - unique cart rule code can be set here, or simply by clicking Generate and Prestashop will generate the rule automatically with unique code name. It will become as voucher from front end side.

Reviewer: ```suggestion
- **Code** - text field - 
If the user decide to use a Code for the cart rule then the code has to be unique to validate the form otherwise a error is thrown: "This cart rule code is already used (conflict with cart rule %Id of the cart rule%).
By clicking Generate and Prestashop will generate the rule automatically with unique code name. It will become as voucher.
If the code is left empty then the cart rule will automatically be applied to customers matching the condition.
```

> - **Highlight switch toggle button** - this option can be enabled, then the cart rule voucher will be displayed in cart summary. When there is no Voucher Code set, the Highlight switch toggle button is hidden.

Reviewer: ```suggestion
- **Highlight** - switch toggle button - If enabled, the cart rule will be displayed in cart summary. 
When there is no Voucher Code set, the Highlight switch toggle button is hidden.
```

> - **Partial use switch toggle button** - this option can be enabled, when the total cart sum is smaller than the voucher sum. If Partial usage is disabled, voucher value will be lovered to the total order amount. If it will be enabled, new voucher will be created in the Cart, as a reminder. Partial use toggle button is switched on by default.

Reviewer: ```suggestion
- **Partial use switch toggle button** - If enabled, when the total cart of the checkout amount is smaller than the cart rule amout, a cart rule is created with the remaining sum.
```
What happens when the action is free shipping or percentage or amount and applied to the order or the specific product while use the partial use feature ?
Is the new cart rule has code? Does the condition match the previous one? 
Does it duplicate the quantity of available? Does the quantity of available is decresead?

Author: @MatShir hey, yes good point, you are right. I have tested, and it becomes a new cart rule with the new cart rule name with "-2" prefix with the remaining sum. The conditions are almost the same, except the cart rule quantity availability - it is just 1 for each user.

> - **Priority** - priority of Cart rules can be set here, in what order to show the rules for the customers of the webshop. For example, setting 1 for the most importand cart rule. Priority input field is stated with digital value 1 by default. If there are multiple cart rules created, with the identical digital value, Prestashop Back-Office will arrange the cart rules like the Cart Rule ID arrangement is set in module Cart Rule listing - from the older to the newest created.

Reviewer: ```suggestion
- **Priority** - Numeric field - By default the priority is set to 1. This field defines the calculation priority of the cart rules at the checkout process. A cart rule with the priority set to 1 will be applied before the cart rule set to 2. The lower the priority value, the higher the priority of the cart rule.
If there are multiple cart rules applied at the checkout with the identical priority value, the priority will be defined between the cart rule by their Cart Rule ID, the lower is the cart rule ID, the higher the priority .
```

> - **Status switch toggle button** - enabling this button will enable the rule, disabling it will deactivate the rule and vouchers for customers. Status toggle button is enabled by default.

Reviewer: ```suggestion
- **Status** - switch toggle button -  Status toggle button is enabled by default. Disabling the toggle will disable the cart rule and won't be useable.
```

## Conditions tab
The condition tab, allows the ability to specify cart rule conditions to determine how the cart rule will be used in the Prestashop. 

[Cart Rules: Conditions Tab](/static/img/cart-rules-conditions-tab.png)

> - **Limit to a single customer input field** - this field sets the customer name or email in order to set the Cart rule condition for certain customer.

Reviewer: ```suggestion
- **Limit to a single custome** - search input - the user can search by customer name or email in order to set the Cart rule condition for one customer.
```

> - **Valid** - simple date picker form, where date existance can be set. Valid data picker widget is stated for the 1 month value by default.

Reviewer: ```suggestion
- **Valid** - 2 dates picker - The date picker "From" and "To" define the period of availability of the cart rule. Past these dates or before these dates, the cart rule can't be used. By default, "From" picker date is set to the date when the form was opened and the "To" one month after the From date.
```

> - **Minimum amount field** - this field sets minimum amount from which the customers will see the Cart rule. As well, the currency can be set, if prices should affected with taxes or without taxes and if shipping is affected too. Minimum amount is stated by digital value 0. Default currency is stated by the default Prestashop currency. Tax excluded dropdown is stated by default. Shipping excluded dropdown is stated by default.

Reviewer: ```suggestion
- **Minimum amount** - this field sets minimum amount from which the customers will can use the cart rule. As well, the currency can be set, if prices should affected with taxes or without taxes and if shipping is affected too. Minimum amount is stated by digital value 0. Default currency is stated by the default Prestashop currency. Tax excluded dropdown is stated by default. Shipping excluded dropdown is stated by default.
```

> - **Total available** - this filed describes the quantity of the cart rules - vouchers available for certain conditions. Total available input field is stated by digital value 1 by default.

Reviewer: ```suggestion
- **Total available** - this filed describes the quantity of the cart rules that can be used. Total available input field is set by default to 1.
```

> - **Total available for each user field** - this field sets the price rules amount for certain users separately. Total available for each user is stated by digital value 1 by default.

Reviewer: What happens if the total available is higher than the total available for each customer? Is the customer can still use the cart rule after using all the total available  for each customer even though there is some total available left?
Can the user still use the cart rule is the total availability is lower than Total available for each user?

Author: @MatShir I have checked, that the error is thrown in that case "This voucher has already been used."

- **Priority** - Numeric field - By default the priority is set to 1. This field defines the calculation priority of the cart rules at the checkout process. A cart rule with the priority set to 1 will be applied before the cart rule set to 2. The lower the priority value, the higher the priority of the cart rule.
If there are multiple cart rules applied at the checkout with the identical priority value, the priority will be defined between the cart rule by their Cart Rule ID, the lower is the cart rule ID, the higher the priority .
- **Status** - switch toggle button -  Status toggle button is enabled by default. Disabling the toggle will disable the cart rule and won't be useable.

## Conditions tab
The condition tab, allows the ability to specify cart rule conditions to determine how the cart rule will be used in the Prestashop. 

![Cart Rules: Conditions Tab](/static/img/cart-rules-conditions-tab.png)

> - **Limit to a single custome** - search input - the user can search by customer name or email in order to set the Cart rule condition for one customer.

Reviewer 2: Limit to a single customer (the r is missing ;) )

- **Minimum amount** - this field sets minimum amount from which the customers will can use the cart rule. As well, the currency can be set, if prices should affected with taxes or without taxes and if shipping is affected too. Minimum amount is stated by digital value 0. Default currency is stated by the default Prestashop currency. Tax excluded dropdown is stated by default. Shipping excluded dropdown is stated by default.
- **Total available** - this filed describes the quantity of the cart rules that can be used. Total available input field is set by default to 1.
- **Total available for each user field** - this field sets the price rules amount for certain users separately. Total available for each user is stated by digital value 1 by default.
### Restrictions
Certain conditions can be set for the main cart rule condition. <br>
Once the first restriction checkbox is marked, the external options will expand. Those are the following:<br>

> - **Carrier selection** - this field can set certain carriers, that could be affected by the rule condition. Once checkboxed, there will be 2 fields prompted in UI - Selected or Unselected text field values. The values can be transferred to appropriate inputs with the buttons Add or Remove. By default, it is set as uncheckboxed condition.

Reviewer 2: The country selection is missing here

![Cart Rules: Actions Tab](/static/img/cart-rules-actions-tab.png)

- **Free shipping switch toggle button** - this setting creates rule that will initiate a free shipping method for customer in checkout form.
- **Apply a discount** - there are 3 options that can be set - Percentage, Amount and None. Percentage can be set, and the cart price will be affected by certain amount of percentage discount, after executing the voucher. When the option is set with the certain Amount - the user will get a discount of a certain amount of price. Also the Currency can be set and the Taxes conditions as well. This option can set the action to None - in that case user of the checkout will not get any discounting at all.
- **Send a free gift switch toggle button** - this option enables ability to set an additional free product for the customer, or every customers, who accepted the cart rule / voucher.
- **Search a product field** - it will set the ability to search the certain product.
- **Matching products dropdown** - it will appear when there will be some characters typed in the "Search a product" field.

> 

Reviewer 2: I think the "Apply the discount to" is missing.

And then, we should say that if the reduction apply to a specific product, then the search a product field appears : 
- **Search a product field** - it will set the ability to search the certain product.
- **Matching products dropdown** - it will appear when there will be some characters typed in the "Search a product" field.

This one should be listed last:
- **Send a free gift switch toggle button** - this option enables ability to set an additional free product for the customer, or every customers, who accepted the cart rule / voucher.
